DependencyInjection/SensioFrameworkExtraExtension.php

```diff
@@ -148,8 +152,13 @@ public function load(array $configs, ContainerBuilder $container)
         if ($config['psr_message']['enabled']) {
             $loader->load('psr7.xml');
 
-            if (!class_exists(ServerRequestFactory::class)) {
-                $definitionsToRemove[] = 'sensio_framework_extra.psr7.argument_value_resolver.server_request';
+            if (!interface_exists(StreamFactoryInterface::class)) {
+                throw new LogicException('PSR-7 support cannot be enabled as the required dependencies are not installed. Try running "composer require nyholm/psr7".');
+            }
+
+            if (class_exists(ServerRequestFactory::class) && !class_exists(Psr17Factory::class)) {
```

Comment:
I would remove this block

Reply:
And everything related to Diactorios to have only Nyholm/PSR-7 as a dev req.

Reply:
If we do that, we need to bump the minimum PHP version to 7.1. Should we do it?

Reply:
Let's do it (and let's bump the version to a new minor)

Reply:
Just while it was green :)
https://travis-ci.org/sensiolabs/SensioFrameworkExtraBundle/builds/504894762
pushed, let's see if it's still green with PHP 7.3 in